tutorials/6 Analysis/Analyzing a transmon and a resonator.ipynb

```diff
@@ -0,0 +1,654 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "horizontal-updating",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "q3d.connect_ansys()\n",
+    "q3d.activate_q3d_design(\"TransmonResonator_q3d\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "caroline-belle",
+   "metadata": {},
+   "source": [
+    "Next, we render the exisitng design to Ansys Q3D for analysis. To ensure that the readout is insulated from the ground plane, we specify the 'open' attribute of render_design to be 'readout'."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "exempt-finnish",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "q3d.render_design(['Q1'], [('Q1','readout')])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "colonial-weight",
+   "metadata": {},
+   "source": [
+    "## Use Ansys Q3D to obtain the capacitance matrix"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "pharmaceutical-steel",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Analyze the default solution setup\n",
+    "q3d.analyze_setup(\"Setup\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "million-temple",
+   "metadata": {},
+   "source": [
+    "For a transmon only case, we should see a 4x4 capacitance matrix, four conductors corresponding to two conducting pads, one connector pad, and the ground palne."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bulgarian-greene",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Using the analysis results, get its capacitance matrix as a dataframe.\n",
+    "q3d.get_capacitance_matrix()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "centered-halifax",
+   "metadata": {},
+   "source": [
+    "The optional keyword parameters of get_capacitance_matrix are given below: <br><br>\n",
+    "variation: str = '' <br>\n",
+    "solution_kind: str = 'AdaptivePass' <br>\n",
+    "pass_number: int = 3"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "crucial-amber",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Run lumped oscillator model (LOM) simulations and save results in dict_lom.b\n",
+    "dict_lom = q3d.lumped_oscillator_vs_passes(12.31, 2, 1, 7.0, [], 9)\n",
+    "dict_lom"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "higher-norman",
+   "metadata": {},
+   "source": [
+    "Using capacitance matrices obtained from each pass, save the many parameters of the Hamiltonian of the system. lumped_oscillator_vs_passes takes in the following parameters: <br><br>\n",
+    "Lj_nH: float <br>\n",
+    "Cj_fF: float <br>\n",
+    "N: int <br>\n",
+    "fr: Union[list, float] <br>\n",
+    "fb: Union[list, float] <br>\n",
+    "maxPass: int <br>\n",
+    "variation: str = '' <br>\n",
+    "solution_kind: str = 'AdaptivePass' <br>\n",
+    "g_scale: float = 1"
```

Comment:
If want, can expand the explanation of the inputs;

    Lj - the Josephson inductance of your Josephson junction 
    Cj - the capacitance of your Josephson junction
    N - the total number of connection pads 
    fr - the frequency of the readout resonator 
    [fb1, fb2, fb3...., fbN-1] - list of the frequencies of the busses
    passes - the number of passes your simulation ran for